### PR TITLE
Status Indicator Improvements

### PR DESCRIPTION
**Pull request by zfox23**
- Added the ability to get a specific user's status from the database
- Added the ability to set custom statuses via an API call (no frontend yet)

### CODE REVIEW

File: usefulUtilities/statusIndicator/hifiScript/statusIndicator.js [hunk 1]
```diff
@@ -167,16 +249,16 @@
...
     }
 
 
     // When avatar goes away, set status to busy
     var previousStatus;
     function onWentAway() {
-        previousStatus = isAvailable;
-        isAvailable = false;
+        previousStatus = currentStatus;
+        currentStatus = false;
```
Comment (robin-k-wilson) [inline usefulUtilities/statusIndicator/hifiScript/statusIndicator.js]: Set to the string "busy

File: usefulUtilities/statusIndicator/hifiScript/statusIndicator.js [hunk 2]
```diff
@@ -221,11 +303,11 @@
         Window.geometryChanged.disconnect(onWindowResize);
         MyAvatar.wentAway.disconnect(onWentAway);
         MyAvatar.wentActive.disconnect(onWentActive);
-        MyAvatar.displayNameChanged.disconnect(sendStatusUpdate);
+        MyAvatar.displayNameChanged.disconnect(updateStatus);
         HMD.displayModeChanged.disconnect(onDisplayModeChanged);
         if (heartbeat) {
             Script.clearTimeout(heartbeat);
-            heartbeat = null;
+            heartbeat = false;
```
Comment (robin-k-wilson) [inline usefulUtilities/statusIndicator/hifiScript/statusIndicator.js]: Do we want all intervals and timeouts to equal "false" when not running as opposed to "null"?

File: usefulUtilities/statusIndicator/hifiScript/statusIndicator.js [hunk 3]
```diff
@@ -29,35 +27,46 @@
...
+        rectangleOverlayProps.x = windowWidth - OVERLAY_WIDTH + OVERLAY_RECTANGLE_LEFT_PADDING - OVERLAY_WINDOW_RIGHT_PADDING;
+        if (currentStatus === "available") {
+            rectangleOverlayProps.color = AVAILABLE_COLOR;
+        } else if (currentStatus === "busy") {
+            rectangleOverlayProps.color = NOT_AVAILABLE_COLOR;
+        } else {
+            rectangleOverlayProps.color = OTHER_COLOR;
+        }
+        rectangleOverlay = Overlays.addOverlay("rectangle", rectangleOverlayProps);
     }
```
Comment (robin-k-wilson) [inline usefulUtilities/statusIndicator/hifiScript/statusIndicator.js]: I'm unable to comment below on lines 77 or 81. When the overlays are unavailable: rectangleOverlay = false; desktopOverlay = false

What do you think?

File: usefulUtilities/statusIndicator/hifiScript/statusIndicator.js (hunk 1 above)
Comment (zfox23) [reply usefulUtilities/statusIndicator/hifiScript/statusIndicator.js]: ty!

File: usefulUtilities/statusIndicator/hifiScript/statusIndicator.js (hunk 2 above)
Comment (zfox23) [reply usefulUtilities/statusIndicator/hifiScript/statusIndicator.js]: developer's choice - i made this `false` to be consistent with the rest of the file :)

File: usefulUtilities/statusIndicator/hifiScript/statusIndicator.js (hunk 3 above)
Comment (zfox23) [reply usefulUtilities/statusIndicator/hifiScript/statusIndicator.js]: ah yes, for consistency, good call

**Comment by salinacsoto** [comment]
bitmoji

**Pull request merged**